### PR TITLE
Constant folding with Quantize

### PR DESCRIPTION
**Pull request by vdevaram**
Added quantization+constant folding . please review

### CODE REVIEW

File: src/ngraph/pass/constant_folding.cpp [hunk 1]
```diff
@@ -529,3 +531,76 @@ void ngraph::pass::ConstantFolding::construct_constant_dequantize()
...
+
+    return make_shared<op::Constant>(quant->get_element_type(), out_shape, out_vec);
+}
+
+void ngraph::pass::ConstantFolding::construct_constant_quantize()
+{
+    auto constant_label =
+        make_shared<pattern::op::Label>(element::f32, Shape{2}, pattern::has_class<op::Constant>());
+    auto q_scale = op::Constant::create(element::f32, Shape{}, {1});
+    auto q_offset = op::Constant::create(element::i8, Shape{}, {1});
```
Comment (diyessi) [inline src/ngraph/pass/constant_folding.cpp]: 0 seems a better default value for offset.

File: src/ngraph/pass/constant_folding.cpp [hunk 2]
```diff
@@ -529,3 +531,76 @@ void ngraph::pass::ConstantFolding::construct_constant_dequantize()
...
+        auto quant_match = pattern_map[quant];
+        auto quantize_op = dynamic_pointer_cast<op::Quantize>(quant_match);
+        auto args = quant_match->get_arguments();
+        auto scale = dynamic_pointer_cast<op::Constant>(args[1]);
+        auto offset = dynamic_pointer_cast<op::Constant>(args[2]);
+
+        auto type = quant_match->get_element_type();
+
+        if (constant_match->get_element_type() != element::f32)
+        {
```
Comment (Krovatkin) [inline src/ngraph/pass/constant_folding.cpp]: do we want check or assert the mode?

File: src/ngraph/pass/constant_folding.cpp [hunk 3]
```diff
@@ -529,3 +531,76 @@ void ngraph::pass::ConstantFolding::construct_constant_dequantize()
...
+        NGRAPH_DEBUG << "In callback for constant_quantize_callback against node = "
+                     << m.get_match_root()->get_name();
+
+        auto pattern_map = m.get_pattern_map();
+
+        auto constant_match = dynamic_pointer_cast<op::Constant>(pattern_map[constant_label]);
+        auto quant_match = pattern_map[quant];
+        auto quantize_op = dynamic_pointer_cast<op::Quantize>(quant_match);
+        auto args = quant_match->get_arguments();
+        auto scale = dynamic_pointer_cast<op::Constant>(args[1]);
```
Comment (Krovatkin) [inline src/ngraph/pass/constant_folding.cpp]: Please use `static_dynamic_cast` here. Matcher makes sure those matched nodes are constant.

File: src/ngraph/pass/constant_folding.cpp (hunk 1 above)
Comment (vdevaram) [reply src/ngraph/pass/constant_folding.cpp]: done

File: src/ngraph/pass/constant_folding.cpp (hunk 2 above)
Comment (vdevaram) [reply src/ngraph/pass/constant_folding.cpp]: checking and returning false

File: src/ngraph/pass/constant_folding.cpp (hunk 3 above)
Comment (vdevaram) [reply src/ngraph/pass/constant_folding.cpp]: done

**Pull request merged**